repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java

```diff
@@ -496,6 +757,42 @@ private boolean incrementAndCheckIfRelationsLimitReached(AtlasEdge atlasEdge, bo
         return hasRelationsLimitReached;
     }
 
+    private boolean incrementAndCheckIfListRelationsLimitReached(AtlasEdge atlasEdge, boolean isInput, AtlasLineageListContext atlasLineageListContext, AtlasLineageListInfo ret, int depth) {
```

Comment:
Need to understand this logic once, LMK when you are free.

Reply:
Sure, lets discuss